src/Skia/Avalonia.Skia/WriteableBitmapImpl.cs

```diff
@@ -31,44 +33,58 @@ public WriteableBitmapImpl(PixelSize size, Vector dpi, PixelFormat? format = nul
             Dpi = dpi;
 
             var colorType = PixelFormatHelper.ResolveColorType(format);
-            
+
             var runtimePlatform = AvaloniaLocator.Current?.GetService<IRuntimePlatform>();
-            
+
             if (runtimePlatform != null)
             {
+                var nfo = new SKImageInfo(size.Width, size.Height, colorType, SKAlphaType.Premul);
+
+                _blob = runtimePlatform.AllocBlob(nfo.BytesSize);
+
+                _pixMap = new SKPixmap(nfo, _blob.Address, nfo.RowBytes);
+
                 _bitmap = new SKBitmap();
 
-                var nfo = new SKImageInfo(size.Width, size.Height, colorType, SKAlphaType.Premul);
-                var blob = runtimePlatform.AllocBlob(nfo.BytesSize);
+                _bitmap.InstallPixels(_pixMap);
+
+                _bitmap.SetImmutable();
```

Comment:
WriteableBitmap is mutable by definition

Reply:
The pixmap is still mutable. If I don't mark the bitmap immutable the pixel data isn't shared.